### PR TITLE
Added Reducers and action

### PR DESCRIPTION
**Pull request by Dmambo**
# Features Added

**Added Redux Toolkit**

[ ] 1.         Wrote  book's actions and reducer  using the duck pattern

**Defined action types for adding and removing a book.**

- [ ]       Set the initial state to be an empty array of books.  
- [ ]       Export Action Creators for the actions.
- [ ]       Wrote reducer and exported it as default.
- [ ]       Define state changes for the actions that was created.
- [ ]       In case of unknown action - return to the current state.
- [ ]       Wrote categories actions and reducer using the ducks pattern.

**Define an action type for checking the status.**
    

- [ ]     Set the initial state to be an empty array of categories.  
- [ ]       Export Action Creators for the actions.
- [ ]       Wrote the reducer and exported it as default.
- [ ]       For the check status action return a string "Under construction".
- [ ]       In case of unknown action - return the current state.
- [ ]       Configure the Redux Store 

- [ ] 
- [ ] Importing the necessary methods from Redux Toolkit.
- [ ] Combine 
- [ ]     both reducers into a root reducer by using configure store function.

### CODE REVIEW

File: src/redux/Books/books.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+const initialState = {
+  books: [],
+};
+
+const bookReducer = createReducer(initialState, (construct) => {
+  construct.addCase('CREATE_BOOK', (state, action) => {
+    state.books.push(action.payload);
+  });
+  construct.addCase('REMOVE_BOOK', (state, action) => {
```
Comment (ha-manel) [inline src/redux/Books/books.js]: - [ ] Kindly use the ducks pattern as mentioned in the project requirements. `CREATE_BOOK` and `REMOVE_BOOK` should be something like `bookstore/books/CREATE_BOOK`... (same for `CHECK_STATUS` in categories) 👍

**Review by ha-manel — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES ♻️

## Hi @Dmambo👋 
Great job so far, There are some changes that need to be implemented in order to proceed to the next project 👍  



## Highlights:  
* All tests passed ✔️
* Professional README file ✔️

## Required Changes:
_Kindly check the comments below_

## Optional:

- [ ] Please consider using best practices when you write commit messages, they should be capitalized sentences in the imperative form, example: "Add linters" instead of "added linters" or "linters are added" ... etc 👍 

------
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/Books/books.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { createReducer, createAction } from '@reduxjs/toolkit';
+
+export const ADD_BOOK = createAction('books/add');
+export const REMOVE_BOOK = createAction('book/remove');
```
Comment (ShadyShawkat) [inline src/redux/Books/books.js]: - [ ] As requested by the previous reviewer, kindly make sure that the actions follow the duck pattern.

- As an example the `ADD_BOOK` action could look more like the following:
```javascript
ADD_BOOK = 'bookstore/books/ADD_BOOK'
```

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @Dmambo, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

Image

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/Books/books.js (hunk 2 above)
Comment (Dmambo) [reply src/redux/Books/books.js]: @ShadyShawkat i am not clear with this please do explain further

Comment (ShadyShawkat) [reply src/redux/Books/books.js]: Hey @Dmambo,

The [duck pattern](https://github.com/erikras/ducks-modular-redux) should look like the following when typing the action types:

```javascript
const ACTION = '<npm-module-or-app/reducer/ACTION_TYPE>'
```

The requested change here is in regard to the string used for every action type

**Review by ShadyShawkat — APPROVED** [review]
## Hi @Dmambo, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- Descriptive commit messages ✔️ 
- Your readme looks professional ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**